core/src/prediction/InstrumentalPredictionStrategy.cpp

```diff
@@ -171,19 +171,26 @@ PredictionValues InstrumentalPredictionStrategy::precompute_prediction_values(
     double sum_YZ = 0;
     double sum_WZ = 0;
 
+    double sum_weight = 0.0;
     for (auto& sample : leaf_samples[i]) {
-      sum_Y += data->get_outcome(sample);
-      sum_W += data->get_treatment(sample);
-      sum_Z += data->get_instrument(sample);
-      sum_YZ += data->get_outcome(sample) * data->get_instrument(sample);
-      sum_WZ += data->get_treatment(sample) * data->get_instrument(sample);
+      sum_Y +=  data->get_weight(sample) * data->get_outcome(sample);
```

Comment:
Maybe we could call `data->get_weight(sample)` once and store it in a variable. I think it would help readability, and wouldn't hurt performance-wise.

Reply:
Done.